pkg/apis/messaging/v1alpha1/channel_validation.go

```diff
@@ -41,9 +42,28 @@ func (cs *ChannelSpec) Validate(ctx context.Context) *apis.FieldError {
 		}
 	}
 
+	if cs.ServiceAccount != nil {
+		err := validateGCPServiceAccount(cs.ServiceAccount)
+		if err != nil {
+			errs = errs.Also(err.ViaField("serviceAccount"))
+		}
+	}
+
 	return errs
 }
 
+func validateGCPServiceAccount(gServiceAccountName *string) *apis.FieldError {
```

Comment:
if you move the method as per the previous comment, I think you won't have to duplicate this code.